sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/functions/ScalarFunction.java

```diff
@@ -30,10 +30,71 @@
  * <p>
  * The JVM type of result values produced by this function must be the type used by Spark's
  * InternalRow API for the {@link DataType SQL data type} returned by {@link #resultType()}.
+ * <p>
+ * <b>IMPORTANT</b>: the default implementation of {@link #produceResult} throws
+ * {@link UnsupportedOperationException}. Users can choose to override this method, or implement
+ * a "magic method" with name {@link #MAGIC_METHOD_NAME} which takes individual parameters
```

Comment:
This conflicts with the doc above which says
```
 * For each input row, Spark will call a produceResult method that corresponds to the
 * {@link #inputTypes() input data types}.
```

Shall we update the doc above?

Reply:
Sure, will update it. It currently says "If no direct method is found or when not using codegen, Spark will call produceResult" which is different from what this PR does, which always call the magic method if it's defined. Do you think it matters much?

Reply:
let's update it to match the actual behavior.